src/com/stanimiriliev/chaptertwo/homogeneoustree/Tree.java

```diff
@@ -18,97 +18,141 @@ public Tree(int root) {
 
```

Comment:
Is that a correct solution? Also I was wondering to remove all getters and setters in the inner class because there is no validations inside them, which would be better?